x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/RestartIndexFollowingIT.java

```diff
@@ -87,16 +92,20 @@ public void testFollowIndex() throws Exception {
         ensureLeaderGreen("index1");
 
         final long thirdBatchNumDocs = randomIntBetween(10, 200);
+        logger.info("Indexing [{}] docs as third batch", thirdBatchNumDocs);
         for (int i = 0; i < thirdBatchNumDocs; i++) {
             leaderClient().prepareIndex("index1").setSource("{}", XContentType.JSON).get();
         }
 
-        assertBusy(
-            () -> assertThat(
-                followerClient().prepareSearch("index2").get().getHits().getTotalHits().value,
-                equalTo(firstBatchNumDocs + secondBatchNumDocs + thirdBatchNumDocs)
-            )
-        );
+        var totalDocs = firstBatchNumDocs + secondBatchNumDocs + thirdBatchNumDocs;
+        assertBusy(() -> {
+            if (hasFollowTaskWithFatalException("index2")) {
+                logger.info("follower has encountered a fatal exception, recreating");
+                assertAcked(followerClient().admin().indices().prepareDelete("index2"));
+                followerClient().execute(PutFollowAction.INSTANCE, putFollow("index1", "index2", ActiveShardCount.ALL)).get();
```

Comment:
Could we pause and unpause the follower instead? Maybe randomly choose between the two

Reply:
I think we can, yes, it should recreate the task and ensure the remote cluster is connected. I randomized the two since recreating is what is documented today.